### DISCUSSION TITLE
If GoAccess could talk, what format would she want server logs in?

### DISCUSSION THREAD

**Discussion by breck7** [allinurl/goaccess#2456] [Q&A] [answered]
I recently switched from nginx to Node.js Express for our primary web server and now we write our own server log (https://github.com/allinurl/goaccess/discussions/2455).

Since we have our own function appending to disk, and all we care about is GoAccess, what's the best format string that we should use?

There are 10 choices listed here: https://github.com/allinurl/goaccess#nearly-all-web-log-formats

Should I use one of those? Or would there be an 11th that is ideal?

**Comment by breck7** [comment, marked as answer]
Seems like the winner is: Combined Log Format (XLF/ELF) Apache | Nginx

**Reply by allinurl** [reply]
I agree. I find the combined log format to be a good format with enough data for basic analytics.

**Reply by allinurl** [reply]
From those 10 options, I'd go with the JSON Structured format. However, if you run apache, I'd use a custom format:

    LogFormat "%v:%p %h %l %u %t \"%r\" %>s %O \"%{Referer}i\" \"%{User-Agent}i\" %D %{Content-Type}o %{SSL_PROTOCOL}x %{SSL_CIPHER}x" vhost_combined_tls
